src/evm/StateChannelSession.ts

```diff
@@ -0,0 +1,22 @@
+import P2pSigner from "./P2pSigner";
+import { AStateMachine } from "@typechain-types";
+import P2pEventHooks from "@/P2pEventHooks";
+
+export class StateChannelSession<T extends AStateMachine> {
+    p2pContractInstance: T;
```

Comment:
Now on the train so have some time to think -> would rename this to P2pInstance - don't like the word "session" since it reminds me of web2 - everything else looks good